tracing-core/src/dispatch.rs

```diff
@@ -404,12 +408,12 @@ where
                     // don't redo this call on the next check
                     *default = get_global().clone();
                 }
-                return f(&*default);
+                return f.take().expect("Failed to get callback (1)")(&*default);
```

Comment:
not a major issue, but i'm not the biggest fan of these panic messages. since they will be visible to the user in the event that we actually _do_ panic here, i think it's important to (a) clearly explain what happened, and (b) state that this is a bug in `tracing-core` and should be reported. maybe something like:
```suggestion
                let f = f.take()
                    .expect(
                        "entered unreachable code: only one branch that calls \
                        `f.take()` should ever execute, so `f` should never \
                        be `None`. this is a bug in `tracing-core`; if you \
                        have encountered this panic, please open a bug report!"
                    );
                return f(&*default);
```